vaadin-portlet/src/main/java/com/vaadin/flow/portal/PortletWebComponentBootstrapHandler.java

```diff
@@ -38,38 +38,36 @@
 class PortletWebComponentBootstrapHandler
         extends WebComponentBootstrapHandler {
 
-    // TODO: Update WebComponentBootstrapHandler API to pass also
-    // current VaadinReponse instance, to avoid having to rely on
-    // VaadinPortletResponse.getCurrentPortletResponse() here.
     @Override
     protected String getServiceUrl(VaadinRequest request,
             VaadinResponse response) {
         final String namespace = ((VaadinPortletResponse) response)
                 .getPortletResponse().getNamespace();
         VaadinPortletSession session = VaadinPortletSession.getCurrent();
-        return VaadinPortlet.getCurrent()
+        return Objects.requireNonNull(VaadinPortlet.getCurrent())
                 .getWebComponentUIDLRequestHandlerURL(session, namespace);
     }
 
     @Override
-    protected String modifyPath(String basePath, String path)
-            throws UnsupportedEncodingException {
+    protected String modifyPath(String basePath, String path) {
         // Require that the static files are available from the server root
         path = path.replaceFirst("^.VAADIN/", "./VAADIN/");
         if (path.startsWith("./VAADIN/")) {
-            DeploymentConfiguration deploymentConfiguration = VaadinPortletService
-                    .getCurrent().getDeploymentConfiguration();
+            VaadinService vaadinService = VaadinPortletService.getCurrent();
+            DeploymentConfiguration deploymentConfiguration =
+                    vaadinService.getDeploymentConfiguration();
+            Optional<DevModeHandler> devModeHandler =
+                    DevModeHandlerManager.getDevModeHandler(vaadinService);
             if (deploymentConfiguration.isProductionMode()
                     || !deploymentConfiguration.enableDevServer()) {
                 // Without dev server we serve static files from the
                 // dedicated URI
                 return getStaticResourcesMappingURI(deploymentConfiguration)
                         + path;
-            } else if (DevModeHandler.getDevModeHandler() != null
-                    && checkWebpackConnection()) {
+            } else if (devModeHandler.isPresent() && checkWebpackConnection()) {
                 // With dev server running request directly from dev server
                 return String.format("http://localhost:%s/%s",
-                        DevModeHandler.getDevModeHandler().getPort(), path);
+                        ((AbstractDevServerRunner) devModeHandler.get()).getPort(), path);
```

Comment:
We should add `getPort` to the `DevModeHandler` interface so that we wouldn't need to bring in the `vaadin-dev-server` dependency just for this.

Reply:
Yes, I'll add a PR to Flow

Reply:
Dev-server dependency excluded.